cwms-data-api/src/main/java/cwms/cda/data/dao/AuthDao.java

```diff
@@ -353,13 +354,16 @@ public void prepareGuestContext(Context ctx) {
         }
     }
 
-    private static String[] getMissingRoles(@NotNull Context ctx,
+    private static List<String> getMissingRoles(@NotNull Context ctx,
                                         @NotNull Set<RouteRole> requiredRoles,
                                         @NotNull DataApiPrincipal p) {
         Set<RouteRole> specifiedRoles = p.getRoles();
         Set<RouteRole> missing = new LinkedHashSet<>(requiredRoles);
         missing.removeAll(specifiedRoles);
-        return missing.stream().map(r -> r.toString()).collect(Collectors.toList()).toArray(new String[0]);
+        return Collections.unmodifiableList(
```

Comment:
FYI this works:
```
return missing.stream()               
.map(Object::toString)               
.collect(Collectors.toUnmodifiableList());
```

Reply:
ugh. thank you, my autocomplete was apparently lying to me. I was sure that had to be a thing.

Reply:
Actually no, my autocomplete was correct, ::toUnmodifiableList was introduced after Java 8.